File: guessing_game.exs
defmodule Game do
  def start do
    random_number = :rand.uniform(200)
    userguess(random_number, 1)
  end
  def userguess(random_number, attempts) do
    guess =
      IO.gets("Enter a Number Between 0-200: ")
      |> String.trim()
      |> String.to_integer()
    cond do
      guess == random_number ->
        IO.puts("You guessed the number in #{attempts} attempts!")
      guess > random_number ->
        IO.puts("Too High")
        userguess(random_number, attempts + 1)
      guess < random_number ->
        IO.puts("Too Low")
        userguess(random_number, attempts + 1)
    end
  end
end
Game.start()

File: guessing_game.php
<?php

resetGame();
function resetGame() {
    $randomNumber = mt_rand(1,100);
    $attempts = 1;
    echo $randomNumber . "\n";
    ask($randomNumber, $attempts);
}

function ask($randomNumber, $attempts) {
    echo "Guess a number between 1 and 100: ";
    $guess = trim(fgets(STDIN));
    identify($guess, $randomNumber, $attempts);
    return $guess;
}

function identify($guess, $randomNumber, $attempts) {
    if ($guess == $randomNumber) {
        echo "Good Job! You got the number in " . $attempts . " attempts! \n";
        playAgain();
    } elseif ($guess < $randomNumber){
        echo "Too Low \n";
        $attempts++;
        ask($randomNumber, $attempts);
    } elseif ($guess > $randomNumber) {
        echo "Too High \n";
        $attempts++;
        ask($randomNumber, $attempts);
    }
}

function playAgain() {
    $stance;
    echo "Play Again? Y/N \n";
    $answer = trim(fgets(STDIN));
    if ($answer == "Y") $stance = true;
    elseif ($answer == "N") $stance = false;
    else {
        echo "Answer Not Acceptable. Please Try Again \n";
        playAgain();
    }
    $stance ? resetGame() : exit;
}
?>